src/Microsoft.AspNet.FileProviders.Combined/CombinedFileProvider.cs

```diff
@@ -0,0 +1,205 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinedFileProvider" /> class using a list of file provider.
+        /// </summary>
+        /// <param name="fileProviders"></param>
+        public CombinedFileProvider(params IFileProvider[] fileProviders)
+        {
+            _fileProviders = fileProviders;
+        }
+
+        /// <summary>
+        /// Locates a file at the given path.
+        /// </summary>
+        /// <param name="subpath">The path that identifies the file. </param>
+        /// <returns>The file information. Caller must check Exists property. This will be the first existing <see cref="IFileInfo"/> returned by the provided <see cref="IFileProvider"/> or a not found <see cref="IFileInfo"/> if no existing files is found.</returns>
+        public IFileInfo GetFileInfo(string subpath)
+        {
+            foreach (var fileProvider in _fileProviders)
+            {
+                var fileInfo = fileProvider.GetFileInfo(subpath);
+                if (fileInfo != null && fileInfo.Exists)
+                {
+                    return fileInfo;
+                }
+            }
+            return new NotFoundFileInfo(subpath);
+        }
+
+        /// <summary>
+        /// Enumerate a directory at the given path, if any.
+        /// </summary>
+        /// <param name="subpath">The path that identifies the directory</param>
+        /// <returns>Contents of the directory. Caller must check Exists property.
+        /// The content is a merge of the contents of the provided <see cref="IFileProvider"/>.
+        /// When there is multiple <see cref="IFileInfo"/> with the same Name property, only the first one is included on the results.</returns>
+        public IDirectoryContents GetDirectoryContents(string subpath)
+        {
+            var directoryContents = new CombinedDirectoryContents(_fileProviders, subpath);
+            return directoryContents;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="IChangeToken"/> for the specified <paramref name="filter"/>.
+        /// </summary>
+        /// <remarks></remarks>
+        /// <param name="pattern">Filter string used to determine what files or folders to monitor. Example: **/*.cs, *.*, subFolder/**/*.cshtml.</param>
+        /// <returns>An <see cref="IChangeToken"/> that is notified when a file matching <paramref name="filter"/> is added, modified or deleted.
+        /// The change token will be notified when one of the change token returned by the provided <see cref="IFileProvider"/> will be notified.</returns>
+        public IChangeToken Watch(string pattern)
+        {
+            // Watch all file providers
+            var activeChangeTokens = new List<IChangeToken>();
+            foreach (var fileProvider in _fileProviders)
+            {
+                var changeToken = fileProvider.Watch(pattern);
+                if (changeToken != null && changeToken.ActiveChangeCallbacks)
```

Comment:
This is the only way I found to determine if a `IChangeToken` is "real" and not a`NoopChangeToken`. I can combine all change tokens, but it will add some overload.